Migration script/openIMIS migration v1.4.2 - 1.5.0.sql

```diff
@@ -1310,3 +1310,362 @@ INSERT [dbo].[tblRoleRight] ([RoleID], [RightID], [ValidityFrom], [ValidityTo],
+		    ,  ISNULL((A.Allocated * (0.01 * @WeightAdjustedAmount) * (0.01 * @ShareContribution))/ NULLIF(SUM(CV.TotalAdjusted) OVER(),0),0) UPAdjustedAmount
+			
+		    FROM tblHF HF
+		    INNER JOIN @HFLevel HL ON HL.Code = HF.HFLevel
+		    LEFT OUTER JOIN tblHFSublevel SL ON SL.HFSublevel = HF.HFSublevel
+		    LEFT JOIN @LocationTemp L ON L.LocationId = HF.LocationId
+		    LEFT OUTER JOIN @TotalPopFam PF ON PF.HFID = HF.HfID
+		    LEFT OUTER JOIN @InsuredInsuree II ON II.HFID = HF.HfID
+		    LEFT OUTER JOIN @InsuredFamilies IFam ON IFam.HFID = HF.HfID
+		   -- LEFT OUTER JOIN @Claims C ON C.HFID = HF.HfID
+		    LEFT OUTER JOIN @ClaimValues CV ON CV.HFID = HF.HfID
+		    LEFT OUTER JOIN @Allocation A ON A.ProdID = @ProdId
+
+		    WHERE HF.ValidityTo IS NULL
+		    AND (((L.RegionId = @RegionId OR @RegionId IS NULL) AND (L.DistrictId = @DistrictId OR @DistrictId IS NULL)) OR CV.ProdID IS NOT NULL OR II.ProdId IS NOT NULL)
+		    AND (HF.HFLevel IN (@Level1, @Level2, @Level3, @Level4) OR (@Level1 IS NULL AND @Level2 IS NULL AND @Level3 IS NULL AND @Level4 IS NULL))
+		    AND(
+			    ((HF.HFLevel = @Level1 OR @Level1 IS NULL) AND (HF.HFSublevel = @Sublevel1 OR @Sublevel1 IS NULL))
+			    OR ((HF.HFLevel = @Level2 ) AND (HF.HFSublevel = @Sublevel2 OR @Sublevel2 IS NULL))
+			    OR ((HF.HFLevel = @Level3) AND (HF.HFSublevel = @Sublevel3 OR @Sublevel3 IS NULL))
+			    OR ((HF.HFLevel = @Level4) AND (HF.HFSublevel = @Sublevel4 OR @Sublevel4 IS NULL))
+		      )
+
+
+
+
+
+	    SELECT  MAX (RegionCode)RegionCode, 
+			MAX(RegionName)RegionName,
+			MAX(DistrictCode)DistrictCode,
+			MAX(DistrictName)DistrictName,
+			HFCode, 
+			MAX(HFName)HFName,
+			MAX(AccCode)AccCode, 
+			MAX(HFLevel)HFLevel, 
+			MAX(HFSublevel)HFSublevel,
+			ISNULL(SUM([TotalPopulation]),0)[Population],
+			ISNULL(SUM(TotalFamilies),0)TotalFamilies,
+			ISNULL(SUM(TotalInsuredInsuree),0)TotalInsuredInsuree,
+			ISNULL(SUM(TotalInsuredFamilies),0)TotalInsuredFamilies,
+			ISNULL(MAX(TotalClaims), 0)TotalClaims,
+			ISNULL(SUM(AlcContriPopulation),0)AlcContriPopulation,
+			ISNULL(SUM(AlcContriNumFamilies),0)AlcContriNumFamilies,
+			ISNULL(SUM(AlcContriInsPopulation),0)AlcContriInsPopulation,
+			ISNULL(SUM(AlcContriInsFamilies),0)AlcContriInsFamilies,
+			ISNULL(SUM(AlcContriVisits),0)AlcContriVisits,
+			ISNULL(SUM(AlcContriAdjustedAmount),0)AlcContriAdjustedAmount,
+			ISNULL(SUM(UPPopulation),0)UPPopulation,
+			ISNULL(SUM(UPNumFamilies),0)UPNumFamilies,
+			ISNULL(SUM(UPInsPopulation),0)UPInsPopulation,
+			ISNULL(SUM(UPInsFamilies),0)UPInsFamilies,
+			ISNULL(SUM(UPVisits),0)UPVisits,
+			ISNULL(SUM(UPAdjustedAmount),0)UPAdjustedAmount,
+			ISNULL(SUM(PaymentCathment),0)PaymentCathment,
+			ISNULL(SUM(TotalAdjusted),0)TotalAdjusted
+	
+	 FROM @ReportData
+
+	 GROUP BY HFCode
+END
+GO
+
+CREATE NONCLUSTERED INDEX NCI_HF_ValidityTo ON tblHF(ValidityTo)
```

Comment:
I miss: 
CREATE TABLE dbo.doc_exa (column_a INT) ;
GO
ALTER TABLE dbo.doc_exa ADD column_b VARCHAR(20) NULL ;
GO
 

And

CREATE NONCLUSTERED INDEX NCI_HF_ValidityTo
ON [dbo].[tblHF] ([ValidityTo])

GO

Reply:
on  Empty databases/openIMIS_OFFLINE.sql  why adding only a single district not a region ?

Reply:
Scope is added in https://github.com/openimis/database_ms_sqlserver/pull/42 
Please review this PR. 

Last line is about index. We don't need a GO for that.

Reply:
Is the link of User with Districts. Locations are added in ONLINE script. Users are linked only with districts. You can see it is tblUsersDistricts.